evaluator/functions.go

```diff
@@ -1161,6 +1165,23 @@ func waitFn(tok token.Token, args ...object.Object) object.Object {
 	return cmd
 }
 
+// kill(`sleep 10 &`)
+func killFn(tok token.Token, args ...object.Object) object.Object {
+	err := validateArgs(tok, "kill", args, 1, [][]string{{object.STRING_OBJ}})
+	if err != nil {
+		return err
+	}
+
+	cmd := args[0].(*object.String)
+
+	if cmd.Cmd == nil {
+		return cmd
+	}
+
+	cmd.Kill()
```

Comment:
I think what we could do in this case, to make sure errors are handled while killing the command, is to return true when the command can be killed, and return an error if we couldn't.

Reply:
👍